File: 课程资料和情报（oj上的AC代码）/今日化学论文.cpp
#include <bits/stdc++.h>
using namespace std;
string tmp;
int where,l;
string dfs(int x){
    string ans="";
    int times=0;
    for(int i=x;i<l;i++){
        if(tmp[i]=='['){
            string cur=dfs(i+1);
            i=where;
            ans+=cur;
        }else if(tmp[i]>='0'&&tmp[i]<='9')times=times*10+(tmp[i]-'0');
        else if(tmp[i]==']'){
            where=i;
            break;
        }else ans+=tmp[i];
    }
    string t=ans;
    for(int i=1;i<times;i++)ans+=t;
    return ans;
}
int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cin>>tmp;
    l=tmp.length();
    cout<<dfs(0);
    return 0;
}

File: 课程资料和情报（oj上的AC代码）/今日化学论文.py
s='['+input()+']'
w=0
def dfs():
    global s,w
    cur,mul='',0
    while w<len(s):
        w+=1
        if s[w-1]=='[':cur+=dfs()
        elif '0'<=s[w-1]<='9':mul=mul*10+int(s[w-1])
        elif s[w-1]==']':return cur*(mul if mul else 1)
        else:cur+=s[w-1]
    return cur*(mul if mul else 1)
print(dfs())
        